In [ ]:
import pandas as pd

price_data = pd.read_csv(r"C:\Users\91739\Downloads\price_data (1).csv")
client_data = pd.read_csv(r"C:\Users\91739\Downloads\client_data (1).csv")
print(price_data.head())
print(client_data.head())


                                 id  price_date  price_off_peak_var  \
0  038af19179925da21a25619c5a24b745  2015-01-01            0.151367   
1  038af19179925da21a25619c5a24b745  2015-02-01            0.151367   
2  038af19179925da21a25619c5a24b745  2015-03-01            0.151367   
3  038af19179925da21a25619c5a24b745  2015-04-01            0.149626   
4  038af19179925da21a25619c5a24b745  2015-05-01            0.149626   

   price_peak_var  price_mid_peak_var  price_off_peak_fix  price_peak_fix  \
0             0.0                 0.0           44.266931             0.0   
1             0.0                 0.0           44.266931             0.0   
2             0.0                 0.0           44.266931             0.0   
3             0.0                 0.0           44.266931             0.0   
4             0.0                 0.0           44.266931             0.0   

   price_mid_peak_fix  
0                 0.0  
1                 0.0  
2                 0.0  
3             

In [2]:
# Data types of each column
print("\n🔍 Price Data - Column Data Types:")
print(price_data.dtypes)

print("\n🔍 Client Data - Column Data Types:")
print(client_data.dtypes)



🔍 Price Data - Column Data Types:
id                     object
price_date             object
price_off_peak_var    float64
price_peak_var        float64
price_mid_peak_var    float64
price_off_peak_fix    float64
price_peak_fix        float64
price_mid_peak_fix    float64
dtype: object

🔍 Client Data - Column Data Types:
id                                 object
channel_sales                      object
cons_12m                            int64
cons_gas_12m                        int64
cons_last_month                     int64
date_activ                         object
date_end                           object
date_modif_prod                    object
date_renewal                       object
forecast_cons_12m                 float64
forecast_cons_year                  int64
forecast_discount_energy          float64
forecast_meter_rent_12m           float64
forecast_price_energy_off_peak    float64
forecast_price_energy_peak        float64
forecast_price_pow_off_peak       float64
has

In [3]:
# Numerical summary
print("\n📊 Price Data - Descriptive Statistics:")
print(price_data.describe())

print("\n📊 Client Data - Descriptive Statistics:")
print(client_data.describe())

# Categorical column analysis
print("\n🔹 Price Data - Unique Values Count:")
print(price_data.nunique())

print("\n🔹 Client Data - Unique Values Count:")
print(client_data.nunique())



📊 Price Data - Descriptive Statistics:
       price_off_peak_var  price_peak_var  price_mid_peak_var  \
count       193002.000000   193002.000000       193002.000000   
mean             0.141027        0.054630            0.030496   
std              0.025032        0.049924            0.036298   
min              0.000000        0.000000            0.000000   
25%              0.125976        0.000000            0.000000   
50%              0.146033        0.085483            0.000000   
75%              0.151635        0.101673            0.072558   
max              0.280700        0.229788            0.114102   

       price_off_peak_fix  price_peak_fix  price_mid_peak_fix  
count       193002.000000   193002.000000       193002.000000  
mean            43.334477       10.622875            6.409984  
std              5.410297       12.841895            7.773592  
min              0.000000        0.000000            0.000000  
25%             40.728885        0.000000            0

In [4]:
# Churn rate analysis
churn_rate = client_data['churn'].mean()
print(f"\n🚀 Client Churn Rate: {churn_rate:.2%}")

# Checking unique IDs
unique_price_ids = price_data['id'].nunique()
unique_client_ids = client_data['id'].nunique()
print(f"🔹 Unique IDs in Price Data: {unique_price_ids}")
print(f"🔹 Unique IDs in Client Data: {unique_client_ids}")



🚀 Client Churn Rate: 9.72%
🔹 Unique IDs in Price Data: 16096
🔹 Unique IDs in Client Data: 14606


Key Observations

1. The churn rate is around 9.7%, meaning ~10% of customers leave.
2. The cons_12m (annual consumption) ranges widely from 0 to 6.2M.
3. The pricing variables have a broad range, indicating different pricing structures.
4. There are 16,096 unique IDs in price data but 14,606 unique IDs in client data 